File: client-clojure/src/sirqul_io_t_platform/specs/user_activity_response.clj
(ns sirqul-io-t-platform.specs.user-activity-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            )
  (:import (java.io File)))


(def user-activity-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :created) inst?
   (ds/opt :latitude) float?
   (ds/opt :longitude) float?
   (ds/opt :tag) string?
   (ds/opt :customId) int?
   (ds/opt :customType) string?
   (ds/opt :userId) int?
   (ds/opt :username) string?
   (ds/opt :appTitle) string?
   (ds/opt :appId) int?
   (ds/opt :appUrl) string?
   (ds/opt :appDescription) string?
   (ds/opt :imageURL) string?
   (ds/opt :profileImage) string?
   (ds/opt :fields) string?
   (ds/opt :id) int?
   (ds/opt :customText) string?
   (ds/opt :customValue) float?
   (ds/opt :customMessage) string?
   (ds/opt :returning) string?
   })

(def user-activity-response-spec
  (ds/spec
    {:name ::user-activity-response
     :spec user-activity-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/UserActivityResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime

import java.time.OffsetDateTime

import UserActivityResponse._

case class UserActivityResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
created: Option[OffsetDateTime],
latitude: Option[Double],
longitude: Option[Double],
tag: Option[String],
customId: Option[Long],
customType: Option[String],
userId: Option[Long],
username: Option[String],
appTitle: Option[String],
appId: Option[Long],
appUrl: Option[String],
appDescription: Option[String],
imageURL: Option[String],
profileImage: Option[String],
fields: Option[String],
id: Option[Long],
customText: Option[String],
customValue: Option[Double],
customMessage: Option[String],
returning: Option[String])

object UserActivityResponse {
  import DateTimeCodecs._

  implicit val UserActivityResponseCodecJson: CodecJson[UserActivityResponse] = CodecJson.derive[UserActivityResponse]
  implicit val UserActivityResponseDecoder: EntityDecoder[UserActivityResponse] = jsonOf[UserActivityResponse]
  implicit val UserActivityResponseEncoder: EntityEncoder[UserActivityResponse] = jsonEncoderOf[UserActivityResponse]
}
